language-server/src/dotty/tools/languageserver/DottyLanguageServer.scala

```diff
@@ -489,10 +489,22 @@ class DottyLanguageServer extends LanguageServer
 
     val uriTrees = driver.openedTrees(uri)
 
-    val defs = Interactive.namedTrees(uriTrees, Include.empty)
+    // Excludes locals from synthetic symbols (with the exception of enums members)
+    // and erroroneous trees.
+    val excludeLocalsFromSyntheticSymbols = (n: NameTree) => {
+      val owner = n.symbol.owner
+      n.symbol.is(Case) || {
```

Comment:
I think this make too many assumptions about which trees to include or not, there might be non-synthetic definitions inside synthetic trees in other places than enums. It might not be enough but as a first step maybe we can just exclude type and term parameters of synthetic definitions:
```suggestion
      !n.symbol.is(Param) || {
```

Reply:
Makes sense. I'm going to commit this soon with the change below.